### PR TITLE
add solution

### PR DESCRIPTION
**Pull request by ssaviak**

### CODE REVIEW

File: src/makeCalculator.js [hunk 1]
```diff
@@ -36,8 +36,39 @@
...
+      return prevValue + number;
+    },
+
+    subtract(prevValue, number) {
+      return prevValue - number;
+    },
+
+    divide(prevValue, number) {
+      return prevValue / number;
+    },
```
Comment (vpashko) [inline src/makeCalculator.js:52]: You can consider that division by zero is not allowed and return an error in such case.

File: src/makeCalculator.js [hunk 2]
```diff
@@ -36,8 +36,39 @@
  *
  * @return {object}
  */
-function makeCalculator() {
-  // write code here
+function makeCalculator(callback, value) {
+  const resObj = {
```
Comment (vpashko) [inline src/makeCalculator.js:40]: [NAMING] - Use proper names for variables. If function was called `makeFruit`, it should return `fruit`. (from checklist)
You can name your object, for example, 'calculator.'

**Review by vpashko — CHANGES_REQUESTED** [review]
Great job! Let's improve your code.

**Review by vpashko — APPROVED** [review]
Good job! Approved

File: src/makeCalculator.js [hunk 3]
```diff
@@ -36,8 +36,44 @@
...
+    subtract(prevValue, number) {
+      return prevValue - number;
+    },
+
+    divide(prevValue, number) {
+      if (number === 0) {
+        throw new Error('Dividing by zero! We keeping previous result');
+      } else {
+        return prevValue / number;
+      }
```
Comment (olya-shyian) [inline src/makeCalculator.js:56]: you can remove this `else` and just return

File: src/makeCalculator.js [hunk 4]
```diff
@@ -36,8 +36,44 @@
...
+    },
+
+    multiply(prevValue, number) {
+      return prevValue * number;
+    },
+
+    reset() {
+      this.result = 0;
```
Comment (olya-shyian) [inline src/makeCalculator.js:64]: this `0`  you can keep in const as a start value

don't forget to use it in `result` as well

**Review by olya-shyian — CHANGES_REQUESTED** [review]
Good job! 🤗 Just minor updates needed

**Review by DanilWeda — APPROVED** [review]
Great!
Approved!

**Pull request closed without merging**